### ISSUE TITLE
`Android Gradle plugin requires Java 11 to run. You are currently using Java 1.8.`  after `ns clean`

### ISSUE THREAD

**Issue by agardes** [NativeScript/NativeScript#9847]
### Issue Description

After deleting the `platforms` folder (android) because of recent issues, when I run `tns debug android` everything is fine until the Gradle Build where I come across this issue : 
```
A problem occurred evaluating root project 'ui_canvas'.
> Failed to apply plugin 'com.android.internal.library'.
   > Android Gradle plugin requires Java 11 to run. You are currently using Java 1.8.
     You can try some of the following options:
       - changing the IDE settings.
       - changing the JAVA_HOME environment variable.
       - changing `org.gradle.java.home` in `gradle.properties`.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
Failed to build plugin @nativescript-community/ui-canvas : 
Error: Command gradlew.bat failed with exit code 1
```

I have the latest Java JDK, and never had this problem before (it's been a while since i didn't do a `tns clean`) : 

```
java version "1.8.0_321"
Java(TM) SE Runtime Environment (build 1.8.0_321-b07)
Java HotSpot(TM) Client VM (build 25.321-b07, mixed mode, sharing)
```

```
echo %JAVA_HOME%  
C:\PROGRA~1\Java\JDK18~1.0_2
```

The only solution I have at the moment is to paste my old `platforms` folder that I thankfully saved. But this is not optimal since it's having issues while building lately.

Here is my package json dependencies : 

```
  "dependencies": {
    "@nativescript-community/insomnia": "^2.0.2",
    "@nativescript-community/tween": "0.0.14",
    "@nativescript-community/ui-canvas": "^4.2.13",
    "@nativescript-community/ui-lottie": "^4.2.0",
    "@nativescript/animated-circle": "^1.1.6",
    "@nativescript/background-http": "^5.0.2",
    "@nativescript/camera": "^5.0.9",
    "@nativescript/core": "~8.0.0",
    "@nativescript/facebook": "^1.0.0-alpha.5",
    "@nativescript/imagepicker": "^1.0.5",
    "@nstudio/nativescript-checkbox": "^2.0.4",
    "@vue/devtools": "^5.3.3",
    "color-convert": "^2.0.1",
    "nativescript-apple-sign-in": "^2.0.0",
    "nativescript-background-http": "^4.2.1",
    "nativescript-imagecropper": "^4.0.3",
    "nativescript-oauth2": "^3.0.7",
    "nativescript-socketio": "^3.3.1",
    "nativescript-toasty": "^3.0.0-alpha.2",
    "nativescript-ui-sidedrawer": "^10.0.2",
    "nativescript-vue": "^2.8.1",
    "nativescript-vue-devtools": "^1.4.0",
    "node-sass": "^4.14.1",
    "vuex": "^3.5.1"
  },
  "devDependencies": {
    "@babel/core": "^7.11.6",
    "@babel/preset-env": "^7.11.5",
    "@nativescript/android": "8.2.2",
    "@nativescript/ios": "8.0.0",
    "@nativescript/webpack": "^3.0.4",
    "babel-loader": "^8.1.0",
    "bezier-js": "^4.0.3",
    "nativescript-vue-template-compiler": "^2.8.1",
    "nativescript-worker-loader": "~0.12.1",
    "sass": "^1.32.8",
    "sass-loader": "^7.3.1",
    "vue-loader": "^15.9.3",
    "webpack-cli": "^3.3.12"
  }
}

```

Thanks a lot if you can provide help!

### Environment

```yaml
OS: Windows 10 10.0.19044
CPU: (16) x64 AMD Ryzen 7 2700 Eight-Core Processor
Shell: Not Found
node: 14.15.1
npm: 8.1.1
nativescript: Not Found

# android
java: Not Found
ndk: Not Found
apis: 24, 28, 30, 31, 32
build_tools: 28.0.3, 30.0.2, 30.0.3
system_images:
  - android-24 | Google APIs Intel x86 Atom
  - android-27 | Google Play Intel x86 Atom
  - android-30 | Google APIs Intel x86 Atom
  - android-30 | Google Play Intel x86 Atom
  - android-31 | Google APIs ARM 64 v8a
  - android-31 | Google Play Intel x86 Atom_64

# ios
xcode: Not Found
cocoapods: Not Found
python: Not Found
python3: Not Found
ruby: Not Found
platforms: Not Found
```

### Dependencies

```json
"dependencies": {
  "@nativescript-community/insomnia": "^2.0.2",
  "@nativescript-community/tween": "0.0.14",
  "@nativescript-community/ui-canvas": "^4.2.13",
  "@nativescript-community/ui-lottie": "^4.2.0",
  "@nativescript/animated-circle": "^1.1.6",
  "@nativescript/background-http": "^5.0.2",
  "@nativescript/camera": "^5.0.9",
  "@nativescript/core": "~8.0.0",
  "@nativescript/facebook": "^1.0.0-alpha.5",
  "@nativescript/imagepicker": "^1.0.5",
  "@nstudio/nativescript-checkbox": "^2.0.4",
  "@vue/devtools": "^5.3.3",
  "color-convert": "^2.0.1",
  "nativescript-apple-sign-in": "^2.0.0",
  "nativescript-background-http": "^4.2.1",
  "nativescript-imagecropper": "^4.0.3",
  "nativescript-oauth2": "^3.0.7",
  "nativescript-socketio": "^3.3.1",
  "nativescript-toasty": "^3.0.0-alpha.2",
  "nativescript-ui-sidedrawer": "^10.0.2",
  "nativescript-vue": "^2.8.1",
  "nativescript-vue-devtools": "^1.4.0",
  "node-sass": "^4.14.1",
  "vuex": "^3.5.1"
},
"devDependencies": {
  "@babel/core": "^7.11.6",
  "@babel/preset-env": "^7.11.5",
  "@nativescript/android": "8.2.2",
  "@nativescript/ios": "8.0.0",
  "@nativescript/webpack": "^3.0.4",
  "babel-loader": "^8.1.0",
  "bezier-js": "^4.0.3",
  "nativescript-vue-template-compiler": "^2.8.1",
  "nativescript-worker-loader": "~0.12.1",
  "sass": "^1.32.8",
  "sass-loader": "^7.3.1",
  "vue-loader": "^15.9.3",
  "webpack-cli": "^3.3.12"
}
```

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

**Comment by agardes** [comment]
Thanks @triniwiz as usual for your quick answer !

I had the latest JDK but indeed I  needed to update Java SE.

Now I get other issue blocking from building gradle : 

```
Webpack compilation complete. Watching for file changes.
Webpack build done!
Updating runtime package.json with configuration values...
Project successfully prepared (android)
Building project...
Gradle build...
         + setting applicationId

Script 'D:\app01\node_modules\nativescript-socketio\platforms\android\include.gradle' line: 7
A problem occurred evaluating script.
Could not find method compile() for arguments [io.socket:socket.io-client:1.0.0, include_74ekwx0axqwfsfw5j97ab418e$_run_closure2$_closure3@7e2be8f4] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.
```

I'm worried cause I never had this issue either before (been working on NS since more than a year now)

**Comment by agardes** [comment]
Downgrading ` "@nativescript/android": "8.2.2",` to `8.0.0` seems to fix the issue mentionned in the last comment.

**Comment by triniwiz** [comment]
That error is due to gradle removing the `compile` syntax , plugins are now required to use `implementation` . The plugin responsible is mine although it's fixed in the `5.x` version, although I think it might break devtools so I published a patch 4.0.3 (since this version should still work for you update that 3.x 😃 )
